diff --git a/src/java/org/apache/commons/jxpath/JXPathContext.java b/src/java/org/apache/commons/jxpath/JXPathContext.java
index 935ebcf..aa809a4 100644
--- a/src/java/org/apache/commons/jxpath/JXPathContext.java
+++ b/src/java/org/apache/commons/jxpath/JXPathContext.java
@@ -380,8 +380,17 @@ import org.apache.commons.jxpath.util.KeyManagerUtils;
  * @version $Revision$ $Date$
  */
 public abstract class JXPathContext {
-    private static volatile JXPathContextFactory contextFactory;
-    private static volatile JXPathContext compilationContext;
+	
+	static {
+		// Initialize in this order:
+		// 1) contextFactory
+	    contextFactory = JXPathContextFactory.newInstance();
+		// 2) compilationContext
+	    compilationContext = JXPathContext.newContext(null);		
+	}
+	
+    private static final JXPathContextFactory contextFactory;
+    private static final JXPathContext compilationContext;
 
     private static final PackageFunctions GENERIC_FUNCTIONS =
         new PackageFunctions("", null);
@@ -435,9 +444,6 @@ public abstract class JXPathContext {
      * @return JXPathContextFactory
      */
     private static JXPathContextFactory getContextFactory () {
-        if (contextFactory == null) {
-            contextFactory = JXPathContextFactory.newInstance();
-        }
         return contextFactory;
     }
 
@@ -646,9 +652,6 @@ public abstract class JXPathContext {
      * @return CompiledExpression
      */
     public static CompiledExpression compile(String xpath) {
-        if (compilationContext == null) {
-            compilationContext = JXPathContext.newContext(null);
-        }
         return compilationContext.compilePath(xpath);
     }
 
